android/ui/src/main/java/com/trafi/ui/RouteSearchTab.kt

```diff
@@ -0,0 +1,151 @@
+package com.trafi.ui
+
+import androidx.compose.foundation.*
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Card
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.ColorFilter
+import androidx.compose.ui.res.vectorResource
+import androidx.compose.ui.tooling.preview.Preview
+import com.trafi.ui.component.internal.RouteSearchTabConstants
+import com.trafi.ui.theme.MaasTheme
+import com.trafi.ui.theme.currentTheme
+import com.trafi.ui.theme.darkColors
+
+@Composable
+private val constants
+    get() = RouteSearchTabConstants(currentTheme)
+
+data class TabItem(
+    val id: String,
+    val icon: String,
+    val active: Boolean,
+    val duration: String?,
+    val price: String?,
+)
+
+@Composable
+fun RouteSearchTab(
+    tabItem: TabItem,
+    isSelected: Boolean,
+    modifier: Modifier = Modifier,
+    onRouteTabClick: () -> Unit,
+) {
+    Card(
+        shape = RoundedCornerShape(constants.borderRadius),
+        border = BorderStroke(constants.borderWidth,
+            color = if (isSelected) constants.borderColorActive else constants.defaultBackgroundColor),
+        modifier = modifier
+            .padding(constants.padding)
+            .widthIn(min = constants.minTabWidth)
+            .heightIn(min = constants.minTabHeight)
+    ) {
+        Box(
+            modifier = modifier
+                .clickable(onClick = onRouteTabClick)
+                .background(if (isSelected) constants.activeBackgroundColor else constants.defaultBackgroundColor)
```

Comment:
Could this background and `Modifier.clickable` be set on the parent `Card`? It supports a `backgroundColor` parameter.

Reply:
@justasm First idea was the same as you described but it does not work like that:

- The problem with clickable is that when you set on card, the clickable effect goes off the card (because card has RounderCornerShape)
- With background problem is with active color, because there is primary color with alpha and when you set it on parent layout (in this case Card) the background looks weird

Reply:
Got it